lib/shoppingList.dart

```diff
@@ -0,0 +1,44 @@
+import 'package:flutter/material.dart';
+import 'products.dart';
+
+class MyShoppingList extends StatefulWidget {
+  List<TextEditingController> MyControllers = [];
+  MyShoppingList(List<TextEditingController> controllers)
+      : MyControllers = controllers;
+
+  @override
+  State createState() => _ShoppingListState(MyControllers);
+}
+
+class _ShoppingListState extends State<MyShoppingList> {
+  List<TextEditingController> _controllers = [];
+
+  void _addProduct() {
+    _controllers.add(TextEditingController());
+    setState(() {});
+  }
```

Comment:
Raczej funkcje zapisuje się na końcu, przerzuciłbym ją między linijkę 43. a 44.
Po poprawkach zamiast _controllers użyj widget.myControllers

```suggestion
  void _addProduct() {
    widget.myControllers.add(TextEditingController());
    setState(() {});
  }
  }
```